wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/AuroraHostListProvider.java

```diff
@@ -361,22 +367,34 @@ private List<HostSpec> processQueryResults(final ResultSet resultSet) throws SQL
       hostMap.put(host.getHost(), host);
     }
 
-    int writerCount = 0;
     final List<HostSpec> hosts = new ArrayList<>();
+    final List<HostSpec> writers = new ArrayList<>();
 
     for (final HostSpec host : hostMap.values()) {
-      if (host.getRole() == HostRole.WRITER) {
-        writerCount++;
+      if (host.getRole() != HostRole.WRITER) {
+        hosts.add(host);
+      } else {
+        writers.add(host);
       }
-      hosts.add(host);
     }
 
+    int writerCount = writers.size();
+
     if (writerCount == 0) {
       LOGGER.severe(
           () -> Messages.get(
               "AuroraHostListProvider.invalidTopology"));
       hosts.clear();
+    } else if (writerCount == 1) {
+      hosts.add(writers.get(0));
```

Comment:
If the wrapper driver we don't have a rule that a writer node should be at position 0. You can just add a writer to the host collection.